cores/esp8266/WString.h

```diff
@@ -32,6 +32,9 @@
 // result objects are assumed to be writable by subsequent concatenations.
 class StringSumHelper;
 
+// class for base64 handling, needs access to setLen()
```

Comment:
This isn't going to pass muster, sorry.

On a practical level, we share this file with ESP32.  On an architectural level, having String and base64 mucking about each others' innards feels like a bad thing to do.  I understand where you're coming from, but there's got to be a better way.

Reply:
I agree with you, I also think the friend access is a mistake. did another variant now.